semgrep-core/src/core/Pattern_match.ml

```diff
@@ -74,7 +74,7 @@ type t = {
   (* metavars for the pattern match *)
   env : Metavariable.bindings;
   (* Lazy since construction involves forcing lazy token lists. *)
-  taint_trace : taint_trace Lazy.t option; [@equal fun _a _b -> true]
+  taint_trace : taint_trace Lazy.t option;
```

Comment:
Please add a comment like "we used to have [@ equal fun _a _ b -> true] here, but this was not correct because  ...
we now rely also on the deriving eq of taint_trace (which itself relies on Parse_info.t eq)"